pkgs/top-level/all-packages.nix

```diff
@@ -22593,11 +22593,12 @@ with pkgs;
     terraform_0_10-full
     terraform_0_11
     terraform_0_11-full
+    terraform_0_12
```

Comment:
Adding new versions is fine, but what about removing old ones?
I don't see terraform maintaining previous releases so we should also limit the amounts of versions.

Reply:
I agree. Since 0.12 is actually functionally different we'll probably want to keep 0.11 around for a while after the fact, but the earlier ones are probably disposable. I do remember @shlevy having some use case for even earlier versions but I don't know if he still does.

Reply:
agreed, deleted a bunch of old versions with associated dependants

Reply:
actually even 0.10 should be safe to delete, there hasn't been any backward-incompatible changes in 0.11

Reply:
Agreed from my standpoint, I'd just make sure that @shlevy doesn't have some weird corner case because I do recall him saying he had a reason to keep the older one around.

The rest of the PR seems fine to me, thanks!

Reply:
OK to delete some of the old ones. We ran into cases where TF was updating quickly and breaking backwards compat but certainly the oldest ones can be removed now. We're using 0.11 currently IIRC.

Reply:
Alright, deleting 0.10 as well